### ISSUE TITLE
Gradients from analytical differentiation

### ISSUE THREAD

**Issue by LeonardoBarneschi** [TUM-DAML/gemnet_pytorch#12]
Dear developers,
I apologize in advance for my question, which may be trivial. I would like to train a differentiable NN model on energies, with gradient from analytic differentiation. I don't want to train directly on gradients, but I would like to get them as energy derivative. Basically I will have my own dataset to train the model on energies, get gradients from the ML model differentiation, and then compare them with a few gradients calculated from very expensive QM calculations. That's indeed why I cannot train directly on gradients, it would be impossible for me to create a sizeable training set.
In case this is possible, could you provide a minimal example or reference? 

Thank you in advance.

Best regards,
Leo

**Comment by gasteigerjo** [comment]
Sure, there's no problem with doing that. The model will predict forces as gradients when you set `direct_forces` in the config to `False`. To ignore the forces during training, you just set `rho_force` in config to 0 or you can adapt the loss function defined [here](https://github.com/TUM-DAML/gemnet_pytorch/blob/master/gemnet/training/trainer.py#L333-L343). Depending on how you code it, you might have to fill the forces with dummy values so you don't get errors.

Note that the model always outputs energies and forces, which is not ideal from a training time perspective. To change that, you'll need to make [this computation](https://github.com/TUM-DAML/gemnet_pytorch/blob/master/gemnet/model/gemnet.py#L604-L608) conditional (pass a parameter to deactivate it or only do it during prediction, not training).

However, I see some issues with your setup. I wouldn't expect the model to do well on predicting forces (energy gradients) without ever having seen them during training. You might want to start with a pretrained model instead and finetune it (train it a bit more) on your dataset.
